src/data_store/data_store_merge_features.cpp

```diff
@@ -41,29 +41,23 @@ data_store_merge_features::data_store_merge_features(generic_data_reader *reader
 
 
 data_store_merge_features::~data_store_merge_features() {
-  MPI_Win_free( &m_win );
 }
 
+void data_store_merge_features::exchange_data() {
+  for (auto store : m_subsidiary_stores) {
+    store->set_shuffled_indices_special(m_shuffled_indices);
```

Comment:
perhaps different name as 'no_exchange'  instead of 'special'

Reply:
I made set_shuffled_indices virtual, and over-rode it in data_store_csv